### PR TITLE
Adds option "overlapBoth" to bedpeOverlap

### PR DESCRIPTION
**Pull request by sweavs111**

### CODE REVIEW

File: cmd/bedPeOverlap/bedPeOverlap.go [hunk 1]
```diff
@@ -125,6 +127,51 @@ func SelectIsBed(bedSelectFile string, bedpeInFile string, overlapThreshold floa
...
+		currOverlaps = interval.Query(selectTree, i.A, "any")
+		if len(currOverlaps) > 0 { //if A, the left side of the input bedPe, overlaps any of the select beds, write the bedPe to output.
+			if overlapThreshold == 0 {
+				bothOverlap = interval.Query(selectTree, i.B, "any")
+				if len(bothOverlap) > 0 {
+					bedpe.WriteToFileHandle(out, i)
+				}
+			} else {
+				found = false
+				for _, j := range currOverlaps {
```
Comment (rimangan) [inline cmd/bedPeOverlap/bedPeOverlap.go:156]: I would pull the i,j,k loop iterator variable declarations up top to cut down on allocations

File: cmd/bedPeOverlap/bedPeOverlap.go [hunk 2]
```diff
@@ -125,6 +127,51 @@ func SelectIsBed(bedSelectFile string, bedpeInFile string, overlapThreshold floa
...
+	inBedPe := bedpe.Read(bedpeInFile)
+	out := fileio.EasyCreate(contactOutFile)
+
+	for _, i := range selectRecords {
+		selectIntervals = append(selectIntervals, i)
+	}
+	selectTree := interval.BuildTree(selectIntervals)
+
+	for _, i := range inBedPe {
+		currOverlaps = interval.Query(selectTree, i.A, "any")
```
Comment (rimangan) [inline cmd/bedPeOverlap/bedPeOverlap.go:147]: Optional, but it might be more descriptive to rename currOverlaps -> aOverlaps and bothOverlap -> bOverlaps

**Review by rimangan — APPROVED** [review]
Some optional additions, LGTM!

**Review by rimangan — APPROVED** [review]

**Review by RavenYL — COMMENTED** [review]
Do you want to address Riley's comments now? I agree with them and think they might be faster to fix now as opposed to later, but it's up to you! I can approve now and you can fix later (and generate another PR) if needed

**Comment by sweavs111** [comment]
Raven, I believe I responded to Riley’s comments and added a new commit but I am terrible at GitHub so it might not show up. I can see it on my end. Sorry for the confusion!!

**Review by RavenYL — APPROVED** [review]

**Comment by RavenYL** [comment]
> Raven, I believe I responded to Riley’s comments and added a new commit but I am terrible at GitHub so it might not show up. I can see it on my end. Sorry for the confusion!!

Oh yes you did! I didn't see that the comments are outdated! Sorry my bad! Approving and merging now!

**Pull request merged**